src/dynamo/models/package.ts

```diff
@@ -0,0 +1,278 @@
+import { DocumentClient } from 'aws-sdk2-types/lib/dynamodb/document_client';
+import { v4 as uuidv4 } from 'uuid';
+
+import { CardDetails } from '../../datatypes/Card';
+import CardPackage, { CardPackageStatus, UnhydratedCardPackage } from '../../datatypes/CardPackage';
+import UserType from '../../datatypes/User';
+import createClient, { QueryInput } from '../util';
+
+const User = require('./user');
+const { cardFromId } = require('../../util/carddb');
+
+const client = createClient({
+  name: 'PACKAGE',
+  partitionKey: 'id',
+  indexes: [
+    {
+      name: 'ByVoteCount',
+      partitionKey: 'status',
+      sortKey: 'votecount',
+    },
+    {
+      name: 'ByDate',
+      partitionKey: 'status',
+      sortKey: 'date',
+    },
+    {
+      name: 'ByOwner',
+      partitionKey: 'owner',
+      sortKey: 'date',
+    },
+  ],
+  attributes: {
+    id: 'S',
+    status: 'S',
+    votecount: 'N',
+    date: 'N',
+    owner: 'S',
+  },
+});
+
+//Using keyof .. provides static checking that the attribute exists in the type. Also its own const b/c inline "as keyof" not validating
+const statusAttr: keyof UnhydratedCardPackage = 'status';
+const ownerAttr: keyof UnhydratedCardPackage = 'owner';
+const keywordsAttr: keyof UnhydratedCardPackage = 'keywords';
+
+const createHydratedPackage = (
+  document: UnhydratedCardPackage,
+  owner: UserType, //TODO: User type
+  cards: CardDetails[],
+): CardPackage => {
+  return {
+    id: document.id!,
+    title: document.title,
+    date: document.date,
+    owner,
+    status: document.status,
+    cards: cards,
+    keywords: document.keywords,
+    voters: document.voters,
+    votecount: document.votecount,
+  } as CardPackage;
+};
+
+const hydrate = async (pack?: UnhydratedCardPackage): Promise<CardPackage | undefined> => {
+  if (!pack) {
+    return pack;
+  }
+
+  const owner = await User.getById(pack.owner);
+  const cards = pack.cards.map((c) => {
+    return cardFromId(c);
```

Comment:
Removed the if statement for c.scryfall_id is set. The cards list stored in dynamo is a list of strings, so in hydration they are never objects